pkg/registry/apis/folders/legacy_storage.go

```diff
@@ -139,9 +138,10 @@ func (s *legacyStorage) Get(ctx context.Context, name string, options *metav1.Ge
 		OrgID:        info.OrgID,
 	})
 	if err != nil || dto == nil {
-		if errors.Is(err, dashboards.ErrFolderNotFound) || err == nil {
-			err = resourceInfo.NewNotFound(name)
+		if err == nil {
```

Comment:
this feels a bit awkward.  should be just have a check outside of this if condition?
```
if dto == nil {
  return nil, apierrors.ToFolderStatusError(dashboards.ErrFolderNotFound)
}
```

Reply:
good idea! updated 👍